### ISSUE TITLE
Fire and room text in state

### ISSUE THREAD

**Issue by AndySky21** [doublespeakgames/adarkroom#356]
I know that it's not that relevant as a saving, but is there a reason because state stores both numeric value and text for fire condition and room temperature? And it's also translated, not in English.
I think those short bits of information could be safely removed.

**Comment by AndySky21** [comment]
Ooook, now that I review the state, there's a lot of strange things out there! It stores pretty much everything, including time delays and income/consumption rates for the whole job group. Is it necessary for the script?

**Comment by Continuities** [comment]
I'm not 100% sure anymore. The state code got majorly refactored to support localization and improve the ad-hoc software architecture I wrote originally (commit 30e7a06) I haven't really been in there since. You may want to ask @LucidCrux, if they're still around.

If you're interested, I'd suggest just messing around to see what happens.

**Comment by skj3gg** [comment]
I'm rust on issue #162, but I believe the temp and fire names have been removed from use in the game just not out of the state, which was the purpose of #162. The entire state could use a nice reorganization.

**Comment by AndySky21** [comment]
Look, I've hated the length of that state since I first tried to save it and it was too long.
As soon as I finish the other 2 changes I have in mind, I'll try to play with that.
I also know that it has to store practically 2 maps and several other things. There should be a documentation file where you could write down what is really needed in state, so that it can be reorganized.

**Comment by Continuities** [comment]
Ultimately, I think this is what's strictly needed to make the game work
(what's needed by the software because it is lazily written - my fault -
could be quite different):

- The full map
- The visible map
- The stores quantities
- The village quantities
- Population and worker assignment
- Perks
- Ship state
- Init state of each screen (whether or not each tab has been unlocked yet)

I don't think I'm missing anything...

**Comment by skj3gg** [comment]
Actually while flipping through `State` I would argue there is a lot more that we would want to keep. Most noticeable pieces to remove though would be in `State.income`.

Might be worth looking into things like http://nodeca.github.io/pako/ to compress the save instread of 33% increase it like base64

**Comment by Continuities** [comment]
Yeah, `State.income` could (should?) be implemented as a value calculated
from the worker assignments instead of stored independently.

**Comment by LucidCrux** [comment]
Sorry, won't be of much help here. It was way too long ago to remember any specifics. I _think_ all I did was move everything to one place, organize a bit, and make access with get/set more consistent instead of a bunch of specific functions. I don't think I touched saving and loading. More of a code-centric refactor instead of anything to do with content.

**Comment by AndySky21** [comment]
I'd add to the list above:
+ Remaining time (in seconds) for last production cycles
+ Remaining cooldown time for cooldown buttons (wood gathering, traps checking, death, fire stoking?)
+ Fire state and/or room state (the fire would be enough, as  room temperature depends on it)
+ Builder state (for the beginning)
+ Seen objects, so that if they're lost they can still be bought.
+ Points so far (and generations?)
+ Current game time 
A lot of things can be desumed instead:
- Outside init depends on wood quantity (if the builder is in "helping" state then the outside has been unlocked)
- World init depends on having a compass
- Ship init depends on having found a ship
Also:
= Map mask could be changed from null/true to 0/1. The state is stored as strings, so I guess it would help shrinking it

**Comment by AndySky21** [comment]
About "seen objects", this could be distinguished on this basis: if stores[thing] exists and it's 0, then the thing has been seen (and subsequently consumed), while if stores[thing] does not exists (undefined), then it hasn't been seen yet and cannot be purchased.

**Comment by chuglydude** [comment]
do you guys need help here

is this issue closed

**Comment by gitupofftheflooranddosomework** [comment]
I think that this isnt an issue, and thus should be closed, to store the fire / temp does not make sense, while going away for a period of time it would be worth noting that a room would get cold
